### PR TITLE
feat(perf): Limit columns on list hosts

### PR DESCRIPTION
**Pull request by chambridge**
# Overview
* This change means the tags column will no longer be selected (it is not used)
* This change means the system_profile_facts column will only be selected when the fields are being selected

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/host_query_db.py [hunk 1]
```diff
@@ -32,6 +32,23 @@
 
 logger = get_logger(__name__)
 
+DEFAULT_COLUMNS = [
+    Host.canonical_facts,
+    Host.id,
+    Host.account,
```
Comment (thearifismail) [inline api/host_query_db.py:38]: Though account is ignored with new implementation, it should be included for debugging RHINENG-9710.

**Review by thearifismail — DISMISSED** [review]

**Review by kruai — DISMISSED** [review]

**Review by kruai — DISMISSED** [review]

File: api/host_query_db.py [hunk 2]
```diff
@@ -44,14 +61,20 @@ def get_all_hosts() -> List:
 def _get_host_list_using_filters(
     all_filters: List, page: int, per_page: int, param_order_by: str, param_order_how: str, fields: List[str]
 ) -> Tuple[List[Host], int, Tuple[str], List[str]]:
+    columns = DEFAULT_COLUMNS
     system_profile_fields = ["host_type"]
     if fields and fields.get("system_profile"):
         additional_fields = ("system_profile",)
         system_profile_fields += list(fields.get("system_profile").keys())
+        columns = list(DEFAULT_COLUMNS).append(Host.system_profile_facts)
```
Comment (vkrizan) [inline api/host_query_db.py:69]: `append()` returns `None`.

**Review by kruai — APPROVED** [review]

**Pull request merged**